### PR TITLE
Add signer legacy spender

### PR DESCRIPTION
**Pull request by romaniy173**
Issues:
1. Change ScriptSpender_Signed  to be proper base class
2. Add ScriptSpender_Signed_* class per address type

### CODE REVIEW

File: cppForSwig/Signer.h [hunk 1]
```diff
@@ -209,12 +209,23 @@ class ScriptSpender
...
 {
-public:
-   ScriptSpender_Signed(const UTXO& utxo)  : ScriptSpender(utxo)
+protected:
+   ScriptSpender_Signed(const UTXO& utxo) : ScriptSpender(utxo)
    { }
    ~ScriptSpender_Signed() override = default;
+public:
+   virtual void setSignedWitnessData(const BinaryData &inputSig, const int itemCount) {}
+   virtual void setSignedScript(const BinaryData& inputSig) {}
```
Comment (sergey-chernikov) [inline cppForSwig/Signer.h:218]: You'd better make them abstract, or throw an "unimplemented" exception in each of them.

File: cppForSwig/Signer.h [hunk 2]
```diff
@@ -244,6 +255,33 @@ class ScriptSpender_P2WPKH_Signed : public ScriptSpender_Signed
...
+
+      bw.put_var_int(signedScript_.getSize());
+      bw.put_BinaryData(signedScript_);
+
+      bw.put_uint32_t(getSequence());
+      serializedInput_ = std::move(bw.getData());
+      return serializedInput_.getRef();
+   }
+
+   void setSignedScript(const BinaryData& inputSig) override {
```
Comment (sergey-chernikov) [inline cppForSwig/Signer.h:278]: formatting

File: cppForSwig/Signer.h (hunk 1 above)
Comment (romaniy173) [reply cppForSwig/Signer.h]: Done

File: cppForSwig/Signer.h (hunk 2 above)
Comment (romaniy173) [reply cppForSwig/Signer.h]: Done

**Pull request merged**